### ISSUE TITLE
Source domains for blacklists aren't whitelisted

### ISSUE THREAD

**Issue by ajp2k13** [pi-hole/pi-hole#1826]
**In raising this issue, I confirm the following:** `{please fill the checkboxes, e.g: [X]}`

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] The issue I am reporting can be *replicated*.
- [X] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/pi-hole/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/pi-hole/issues)).

**How familiar are you with the the source code relevant to this issue?:**

1

---
**Expected behaviour:**

Pi-hole is supposed to add the source domains for the blacklists

**Actual behaviour:**

Source domains for the blacklists aren't added.

[✓] Adding 6 blocklist source domains to the whitelist
  [✓] Whitelisting 0 domains
  [✓] Parsing domains into hosts format
  [✓] Cleaning up stray matter

**Steps to reproduce:**

Update lists, check whitelist.

**Debug token provided by [uploading `pihole -d` log](https://discourse.pi-hole.net/t/the-pihole-command-with-examples/738#debug):**

qnd64via0f

Tried reinstalling several times, same thing.

**Comment by dschaper** [comment]
Can you confirm that the source domains are not in `/etc/pihole/white.list`?

**Comment by dschaper** [comment]
I see from the debug that indeed the white.list is empty.

**Comment by ajp2k13** [comment]
I have a file called /etc/pihole/whitelist.txt but not a white.list?

**Comment by Mcat12** [comment]
Yes, the file is called `whitelist.txt`. What is the output of these commands (the last one will give a token so we can securely see the debugging log of gravity log):
```
sudo bash -x /opt/pihole/gravity.sh &> gravity.log
cat gravity.log | pihole tricorder
```

**Comment by ajp2k13** [comment]
Here's the token:

3khigyrrb9

Thanks!

**Comment by Mcat12** [comment]
Sorry, I was unable to get to your gravity log before it expired (48 hour limit). Can you make a new token?

**Comment by ajp2k13** [comment]
No worries, here's the new token...

ps2kfzbu6f

Thanks!

**Comment by Mcat12** [comment]
What is the contents of `/etc/pihole/whitelist.txt` before and after running this command (and please provide the output of the command)?
```
pihole -w -nr -q 'raw.githubusercontent.com mirror1.malwaredomains.com sysctl.org zeustracker.abuse.ch s3.amazonaws.com hosts-file.net
```

**Comment by ajp2k13** [comment]
whitelist.txt is empty but I can't get the command to run? It just sits there...

pi@pihole:~ $ pihole -w -nr -q 'raw.githubusercontent.com mirror1.malwaredomains.com sysctl.org zeustracker.abuse.ch s3.amazonaws.com hosts-file.net
>

**Comment by Mcat12** [comment]
Very odd... Then when you run gravity does it work?

**Comment by ajp2k13** [comment]
It would appear so, now when I run "pihole -g" I get this...


  [✓] Adding 6 blocklist source domains to the whitelist
  [✓] Whitelisting 6 domains
  [✓] Parsing domains into hosts format
  [✓] Cleaning up stray matter

**Comment by Mcat12** [comment]
Hmm, if you remove those same domains from the whitelist (with `pihole -w -d`) and then run gravity what happens?

**Comment by ajp2k13** [comment]
I removed all the the domains from the whitelist, using pihole -w -d for each domain. Then I ran pihole -g... weird?

  [✓] Adding 6 blocklist source domains to the whitelist
  [✓] Whitelisting 0 domains
  [✓] Parsing domains into hosts format
  [✓] Cleaning up stray matter

**Comment by Mcat12** [comment]
Ok, so to debug further it would be helpful to see the exact output when a certain line is run in the script.
So in `/opt/pihole/gravity.sh` edit this line:
https://github.com/pi-hole/pi-hole/blob/master/gravity.sh#L422
and remove the `&> /dev/null` at the end. Then run `pihole -g` again and share the output.

**Comment by ajp2k13** [comment]
Sorry but I can't find that line anywhere...

**Comment by Mcat12** [comment]
You can use `vim` to open the file as `root`, then press `:set number` to show the line numbers. Or if you have a desktop installed you can use a GUI file editor as long as it's run through root.

**Comment by ajp2k13** [comment]
Thanks, found it!

Output (skipped the first part)

[✓] Consolidating blocklists
  [✓] Extracting domains from blocklists
  [i] 129,693 domains being pulled in by gravity
  [✓] Removing duplicate domains
  [i] 106,541 unique domains trapped in the Event Horizon

  [i] Adding 6 blocklist source domains to the whitelist...  [✗] raw.githubusercontent.com mirror1.malwaredomains.com sysctl.org zeustracker.abuse.ch s3.amazonaws.com hosts-file.net is not a valid argument or domain name!
  [✓] Adding 6 blocklist source domains to the whitelist
  [✓] Whitelisting 0 domains
  [✓] Parsing domains into hosts format
  [✓] Cleaning up stray matter

**Comment by Mcat12** [comment]
Hmm... Try removing the `"` around `${uniqDomains[*]}` and try again.

**Comment by ajp2k13** [comment]
Removed the " around ${uniqDomains[*]}

 [✓] Consolidating blocklists
  [✓] Extracting domains from blocklists
  [i] 129,693 domains being pulled in by gravity
  [✓] Removing duplicate domains
  [i] 106,541 unique domains trapped in the Event Horizon

  [i] Adding 6 blocklist source domains to the whitelist...
  [i] Using cached Event Horizon list...
  [i] 106,541 unique domains trapped in the Event Horizon
  [✓] Whitelisting 6 domains
  [✓] Parsing domains into hosts format
  [✓] Cleaning up stray matter

  [✓] Force-reloading DNS service
  [✓] DNS service is running
  [✓] Pi-hole blocking is Enabled
  [✓] Adding 6 blocklist source domains to the whitelist
  [✓] Whitelisting 6 domains
  [✓] Parsing domains into hosts format
  [✓] Cleaning up stray matter

  [✓] Force-reloading DNS service
  [✓] DNS service is running
  [✓] Pi-hole blocking is Enabled

**Comment by ajp2k13** [comment]
Sure thing, here it is... :)

h9es73v4ix

**Comment by PromoFaux** [comment]
Cheers. Actually, I misunderstood this entire issue and thought that I was unable to reproduce it _(Hint:- try not to be hungover when looking through issues)_

Looks like removing the `"`'s around that array variable fixes it.

**Comment by ajp2k13** [comment]
Hope it's not too bad... :)

So I'm not the only one experiencing this issue then?

**Comment by PromoFaux** [comment]
Looking at it, the bug has been there since at least v3.1.4, and nobody else has brought it up!

It's one of those things, we only add the source domains to the whitelist as a precaution to stop lists from blocking each other, and you would probably only notice it if one of your lists couldn't download.

Thanks for being eagle-eyed enough to notice they weren't being added! We can hang our collective dev-heads in shame ;)

**Comment by ajp2k13** [comment]
Easy enough to miss, I only noticed it because I reinstalled my Pi-hole completely when 3.2 came out. :)

Thanks for the friendly support, you guys rock!

**Comment by dschaper** [comment]
Closed via #1856
